lib/rucio/common/utils.py

```diff
@@ -1150,3 +1175,59 @@ def get_parsed_throttler_mode(throttler_mode):
         direction = 'source'
         all_activities = True
     return (direction, all_activities)
+
+
+def filter_thread_load(query, column_name, total_workers, worker_number, session=None):
```

Comment:
The 1.21.6 version released today now introduces exactly this method, I think it's even called the same. 😄 Please switch your code to that one and remove yours here.

Reply:
It is called filter_thread_work and I am happy to delete few lines of my code thanks to it ! Done ;o)